File: src/day10/main.py
def p1(input):
    x = 1
    xs = [1,1]
    for line in input.splitlines():
        xs.append(x)
        if line.startswith("addx"):
            x += int(line[5:])
            xs.append(x)
    result = sum((i) * xs[i] for i in range(20,221, 40))
    print(f"part 1: {result}")
    return xs

def p2(xs):
    print(f"part 2: ")
    for y in range(6):
        for x in range(40):
            sprite = xs[40*y+x]
            if 0 <= (x-sprite) <= 2:
                print('[]', end='')
            else:
                print('  ', end='')
        print()

def main(input):
    xs = p1(input)
    p2(xs)

File: src/day10/main.rs
use shared::puzzle_input;

pub fn main() {
    let input = puzzle_input!();
    let xs = input.split('\n').fold(vec![1, 1], |mut xs, line| {
        let x = *xs.last().unwrap();
        xs.push(x);
        if line.starts_with("addx") {
            let v = line[5..].parse::<isize>().unwrap();
            xs.push(x + v);
        }
        xs
    });
    let p1: isize = (20..=220).step_by(40).map(|i| i as isize * xs[i]).sum();
    print!("part 1: {}\npart 2: \n", p1);
    for y in 0..6 {
        for x in 0..40 {
            let sprite = xs[(y * 40 + x) as usize];
            let s = if (0..=2).contains(&(x - sprite)) {
                "[]"
            } else {
                "  "
            };
            print!("{}", s);
        }
        println!();
    }
}
